backend-rust/src/graphql_api/block_metrics.rs

```diff
@@ -108,14 +108,37 @@ LIMIT 1"#,
             .as_duration()
             .try_into()
             .map_err(|err| ApiError::DurationOutOfRange(Arc::new(err)))?;
+
         let period_query = sqlx::query!(
-            r#"
-SELECT
-    COUNT(*) as blocks_added,
-    AVG(block_time)::integer as avg_block_time,
-    AVG(finalization_time)::integer as avg_finalization_time
-FROM blocks
-WHERE slot_time > (LOCALTIMESTAMP - $1::interval)"#,
+            "WITH
+                 p_start AS (
+                     SELECT
+                         height,
+                         slot_time,
+                         cumulative_finalization_time
+                     FROM blocks
+                     WHERE (LOCALTIMESTAMP - $1::interval) <= slot_time
+                     LIMIT 1
+                 ),
+                 p_end AS (
+                     SELECT
+                         height,
+                         slot_time,
+                         cumulative_finalization_time
+                     FROM blocks
+                     ORDER BY slot_time DESC
+                     LIMIT 1
+                 )
+             SELECT
+                 p_end.height - p_start.height AS blocks_added,
+                 ((p_end.slot_time - p_start.slot_time) /
+                     NULLIF(p_end.height - p_start.height, 0)
+                 ) AS avg_block_time,
+                 ((p_end.cumulative_finalization_time - \
+             p_start.cumulative_finalization_time)::float /
+                     NULLIF(p_end.height - p_start.height, 0) * 1000
+                 ) AS avg_finalization_time
```

Comment:
It seems as if we are anyways mapping this value. Could we extend it with the unit? (_ms)

Reply:
I have added units now.
These are actually converting to seconds as part of the query now, so removed the mapping :+1: 
Good thing you brought it up